install/kubernetes/istio-gateway.yaml

```diff
@@ -0,0 +1,55 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: istio-gateway
+spec:
+  selector:
+    istio: ingressgateway # use istio default controller
+  servers:
+  - port:
+      number: 15007
+      name: pilot-http-discovery
+      protocol: TCP
+    hosts:
+    - "*"
+  - port:
+      number: 15010
+      name: pilot-grpc-xds
+      protocol: TCP
+    hosts:
+    - "*"
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: pilot-http-discovery
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - istio-gateway
+  tcp:
+  - match:
+    - port: 15007
+    route:
+    - destination:
+        host: istio-pilot.istio-system.svc.cluster.local
+        port:
+          number: 15007
+---
+kind: VirtualService
+metadata:
+  name: pilot-grpc-xds
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - istio-gateway
+  tcp:
+  - match:
+    - port: 15010
+    route:
```

Comment:
Collapse all of these into a single virtual services and have multiple match rules within the virtual service. In fact, you could do something as simple as
```yaml
- match:
  - port: 1
  route:
   - destination: 
       host: ...
       port: port1
  - port: 2
  - port: 3
  route: # common across all 3 port matches
    ...
```

I "think" you can even omit the port in the route because it will be auto inferred based on the match port/listener port. If it doesn't work, file an issue. We have this feature for http.

Reply:
you are right, the port can be omitted in the route.